### ISSUE TITLE
Mac Copying "-" instead of the text to be copied.

### ISSUE THREAD

**Issue by akashp1997** [jswanner/DontFuckWithPaste#24]
The chrome on mac is not copying the content but instead is copying just a dash character or "-".

**Comment by jswanner** [comment]
@akashp1997 Can you tell me on what site you are experiencing this behavior? I have seen anything like that before.

Also, you might want to open the options for the extension and add the site that's giving you trouble to the list of excluded sites.

**Comment by akashp1997** [comment]
Alright, so it does not happen only in mac. It happens on all the OSs. The issue is with the character limit perhaps. I tried copying and pasting in short portions and it worked. I think removing this character limit will fix this issue. The website is an internal college website which cannot be accessed online.

**Comment by jswanner** [comment]
If you completely disable this extension does the problem go away?

**Comment by jswanner** [comment]
By the way, there is no character limit in this extension, and no part of it inspects the content of selected text nor the clipboard contents.

The majority of the code in this extension deals with settings, as far as the code that actually deals with copying and pasting events, it's just: https://github.com/jswanner/DontFuckWithPaste/blob/d8ecb4c646fe755de8348538b509a38d2a64fe35/content.js#L2-L3.

**Comment by akashp1997** [comment]
Alright, with vigorous checking, I found that there is a character limit of 1000 on the website I am checking it on. As soon as 1001 characters are selected for copying, a dash character or "-" is seen on the clipboard. I hope I could help you with this but I am not that familiar with JS.

**Comment by guiambros** [comment]
I was experiencing the same, and realized this is caused by an incompatibility between [CodeMirror](https://codemirror.net/) (a very popular web text editor) and DFWP extension, as explained [here](https://github.com/Codewars/codewars.com/issues/1024#issuecomment-314312415).

*TL;DR*: you have to exclude any sites that use CodeMirror's text editor. This includes Jupyter Notebook, CodeWars, and [many others](http://codemirror.net/3/doc/realworld.html).

**Comment by jswanner** [comment]
For backwards compatibility with older versions of this extension, the default configuration of this extension is to blacklist all sites and whitelist a few. I think the opposite is a better solution: whitelist all sites and blacklist only the troublesome ones.

I have a complete rewrite of this extension to do just that, it's working, but needs UX love. If you are familiar with using local extension, you can test it out: https://github.com/jswanner/DontFuckWithPaste/tree/v2.
